### DISCUSSION TITLE
Feature Request: Add support for CRSF/CrossFire 'Packed Channels' to allow 13bit resolution instead of 11bit

### DISCUSSION THREAD

**Discussion by hydra** [ExpressLRS/ExpressLRS#2086] [Ideas]
After some poking around today in the BF codebase, and referencing the official TBS Crossfire Protocol documentation (Revision 10) it seems that Betaflight can accept the 'Packed Channels' packets which can contain up to 13 bits-per-channels, ELRS current uses the `0x16 RC_CHANNELS_PACKED` frame which only supports 11bits per channel.

The `0x17 SUBSET_RC_CHANNELS_PACKED` frame support 10-13 bits.

Here's the links to the code in ELRS which creates the frames:

https://github.com/ExpressLRS/ExpressLRS/blob/c5f67bd28a9f559a0bba3a4bcf509a4340445df5/src/lib/CRSF/CRSF.cpp#L993-L1033

Here's the code in BF that handles the frames:
https://github.com/betaflight/betaflight/blob/1c92d83c6fdd15c7a7700de2ee2d61768a7a415e/src/main/rx/crsf.c#L390-L395
https://github.com/betaflight/betaflight/blob/1c92d83c6fdd15c7a7700de2ee2d61768a7a415e/src/main/rx/crsf.c#L485-L570

Here's where the 13 bits of channel data are used:
https://github.com/betaflight/betaflight/blob/1c92d83c6fdd15c7a7700de2ee2d61768a7a415e/src/main/rx/crsf.c#L543-L547

from Discord, JamesK / @JBKingdon noted that using a higher baud rate may be required at high ELRS packet rates.

The TBS spec says that `0x17 SUBSET_RC_CHANNELS_PACKED` packets are only sent after baud rate negotiation, but if I'm reading the BF code correctly above BF will always accept those frames it's handled right next to the `0x16 RC_CHANNELS_PACKED` frame.

**Comment by JyeSmith** [comment]
Im interested at this and the FLRC OTA packets have room for 12b channels.  These are the faster rates where higher bit channels make sense.  But first we need the handset to provide high res channel data e.g. sample at 13b and calibrate/endpoints to good 12b data.

**Comment by CapnBry** [comment]
Just to clarify since people seem to be finding this: "Packed Channels" is what we already have, subset channels is I guess what you call the CRSFv3 stuff.

The MAJOR downside of CRSFv3 is it is limited to -100% to +100%, which sucks donkeys for everything except quads. EdgeTX supports +/-150% and any CRSF protocol update that throws out the current +/-121% CRSF range is a half step backward imo. More precision and more range would be ideal and this reduces range while increasing precision only.

Just doing 13bit +/-150% covers both cases as that's 0.037% per bit across the full range, more than 12bit covering just +/-100% (0.048% per bit). You really wouldn't want to separate them into 4x 13 bit channels (because just sticks lol) + 12x lower res channels for the minimal transfer time gain difference at like 5Mbps when you consider people use more than 4 channels and those people will whine endlessly that they can feel that one aileron moves in 13 bit and the other feels more like 11 bit. 😆
